pkg/crd/gen.go

```diff
@@ -201,9 +201,14 @@ func removeDefaultsFromSchemaProps(v *apiextlegacy.JSONSchemaProps) {
 		return
 	}
 
+	if v.Default != nil {
+		fmt.Println("Warning: default unsupported in CRD version v1beta1, v1 required. Removing defaults.")
+	}
+
 	// nil-out the default field
 	v.Default = nil
 	for name, prop := range v.Properties {
+		prop := prop
```

Comment:
add a comment that notes that this is to avoid taking a reference to the iteration variable (or just add the comment and a `//nolint:gosec` -- we intentionally handle the persistence of the modifications on the line below)